src/main/java/org/spdx/sbom/gradle/utils/SpdxDocumentBuilder.java

```diff
@@ -282,9 +282,12 @@ private SpdxPackage createProjectPackage(ResolvedComponentResult resolvedCompone
               + " has no specified version");
       version = "NOASSERTION";
     }
-    var supplier = documentInfo.getSupplier().orElse("NOASSERTION");
-    if (supplier.equals("NOASSERTION")) {
-      logger.warn("supplier not set for project " + pi.getName());
+    var supplier = "NOASSERTION";
+    if (pi == describesProject || pi.getRootName().equals(describesProject.getRootName())) {
```

Comment:
I think we should still inspect the supplier if it's available no matter what?
I would expect something like for this logic to be:
```java
var supplier = documentInfo.getSupplier().orElse("NOASSERTION");
if (supplier.equals("NOASSERTION") && (pi == describesProject || pi.getRootName().equals(describesProject.getRootName())) {
    logger.warn("supplier not set for project " + pi.getName());
}
```
But I'm not sure I understand why the shared root matters? What does the condition `pi.getRootName().equals(describesProject.getRootName())` provide?

Reply:
oh wait, I misunderstood what was going on here. I still don't think I understand the common root condition though.

Reply:
how can a project be constructed where there is no common root?

Reply:
I have to double-check this, I've had a problem with supplier not being set properly for external artifacts (e.g. maven dependencies), but now I think it comes from pom parsing

Reply:
yeah all external artifacts get their supplier from the POM

Reply:
After rebasing fork, everything seem to be working as expected